external/model-preparation-algorithm/mpa_tasks/extensions/datasets/mpa_cls_dataset.py

```diff
@@ -238,3 +262,124 @@ def evaluate(self,
                     eval_results[k] = v
 
         return eval_results
+
+
+@DATASETS.register_module()
+class MPAHierarchicalClsDataset(MPAMultilabelClsDataset):
+    def __init__(self, **kwargs):
+        self.hierarchical_info = kwargs.pop('hierarchical_info', None)
+        super().__init__(**kwargs)
+
+    def load_annotations(self):
+        include_empty = self.empty_label in self.labels
+        for i, _ in enumerate(self.ote_dataset):
+            class_indices = []
+            item_labels = self.ote_dataset[i].get_roi_labels(self.labels, include_empty=include_empty)
+            ignored_labels = self.ote_dataset[i].ignored_labels
+            if item_labels:
+                num_cls_heads = self.hierarchical_info['num_multiclass_heads']
+
+                class_indices = [0]*(self.hierarchical_info['num_multiclass_heads'] +
+                                     self.hierarchical_info['num_multilabel_classes'])
+                for j in range(num_cls_heads):
+                    class_indices[j] = -1
+                for ote_lbl in item_labels:
+                    group_idx, in_group_idx = self.hierarchical_info['class_to_group_idx'][ote_lbl.name]
+                    if group_idx < num_cls_heads:
+                        class_indices[group_idx] = in_group_idx
+                    else:
+                        if ote_lbl not in ignored_labels:
```

Comment:
This part also seems to need to be modified to 'elif'.

Reply:
Revised. Thanks.